rest/routes/user.py

```diff
@@ -122,24 +120,24 @@ def admin_editor_requests():
         ]
     :return: success status and json editor requests (if GET)
     """
-    if request.method == 'OPTIONS':
-        response = Response(response=json.dumps({}), status=200, mimetype='application/json')
-        response.headers.add('Access-Control-Allow-Origin', '*')
-        response.headers.add('Access-Control-Allow-Headers', 'email, content-type')
-        return response
+    if 'token' not in request.headers:
+        return create_response({'message': 'No token found, log in!'}, 401, '*')
 
-    if 'email' not in request.headers:
-        response = Response(response=json.dumps({}), status=401, mimetype='application/json')
-        response.headers.add('Access-Control-Allow-Origin', '*')
-        return response
+    try:
+        token = jwt.decode(
+            request.headers['token'],
+            current_app.config['SECRET_KEY'],
+            algorithms='HS256'
+        )
+    except jwt.ExpiredSignatureError:
+        return create_response({'message': 'Token expired, log in again!'}, 401, '*')
+    except jwt.InvalidSignatureError:
+        return create_response({'message': 'Invalid token signature!'}, 401, '*')
```

Comment:
What about moving jwt decoding to before request function? We need to check token in every endpoint except /user POST and /user/auth POST, so there could be a lot of code duplication.

Reply:
There is no way of using return value from before request function as when it has non-None value return further handling of the request is stopped. So there would have to be some hardcoded dictionary of urls where you check for permission (i.e. '/user/isEditor/': 'editor') and look for current request url in the dictionary.

I think better idea is to create function in rest/common wchich returns role or Response if an error with the token occurred. And then a route handles what to do with the role its gotten or returns the Response.

Reply:
But when the jwt is invalid we should't handle request further. All we had to hardcode would be just two urls that don't need authorization (/users POST, /users/auth POST) to omit the jwt validation check if request is for one of those endpoints.

However your solution is also correct so, it can stays how it is right now :)